atlasdb-commons/src/main/java/com/palantir/async/initializer/AsyncInitializer.java

```diff
@@ -71,7 +75,6 @@ private void tryInitializationLoop() {
                     SafeArg.of("className", getInitializingClassName()),
                     SafeArg.of("numberOfAttempts", numberOfInitializationAttempts),
                     SafeArg.of("initializationDuration", System.currentTimeMillis() - initializationStartTime));
-            singleThreadedExecutor.shutdown();
```

Comment:
can we ever shut this down?

Reply:
Maybe you could close all of them when `SerializableTransactionManager` is initialized, as you don't expect any of them to be closed in the middle of a initialization anymore... but I'm not sure. @nziebart for SA.

Reply:
would be nice to have a test for this

Reply:
also - wonder if we should add a try/catch around `cleanupOnInitFailure`. If something throws in there, currently we would just die and never initialize

Reply:
perhaps `try {cleanupOnInitFailure} finally {scheduleInitialization()}`, so we at least always schedule?